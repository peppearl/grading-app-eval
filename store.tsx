import create from "zustand";

export const useStore = create((set) => ({
    notes: [],
    addNote: (title, content, comment) => {
        const newNote = {
            title,
            content,
            comment,
            created: new Date().toLocaleString(),
        };
        set((state) => ({ notes: [...state.notes, newNote] }));
    },
    deleteNote: (index) => {
        set((state) => {
            const updatedNotes = [...state.notes];
            updatedNotes.splice(index, 1);
            return { notes: updatedNotes };
        });
    },
    updateNote: (index, title, content, comment) => {
        set((state) => {
            const updatedNotes = [...state.notes];
            updatedNotes[index].title = title;
            updatedNotes[index].content = content;
            updatedNotes[index].comment = comment;
            return { notes: updatedNotes };
        });
    },
}));