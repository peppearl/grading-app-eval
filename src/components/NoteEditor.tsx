import {useStore} from "../../store";

export const NoteEditor = ({ index, title, content, comment, setShowEditor }) => {
    const updateNote = useStore((state) => state.updateNote);

    const handleUpdateNote = () => {
        updateNote(index, title, content, comment);
        setShowEditor(false);
    };

    const handleContentChange = (e) => {
        const value = e.target.value;
        // Validate input to only allow numbers between 0 and 20
        if (value === '' || (value >= 0 && value <= 20)) {
            updateNote(index, title, value, comment);
        }
    };

    return (
        <div>
            <input
                type="text"
                value={title}
                onChange={(e) => updateNote(index, e.target.value, content, comment)}
            />
            <input
                type="number"
                value={content}
                onChange={handleContentChange}
            />
            <textarea
                value={comment}
                onChange={(e) => updateNote(index, title, content, e.target.value)}
            ></textarea>
            <button onClick={handleUpdateNote}>Sauvegarder</button>
        </div>
    );
};