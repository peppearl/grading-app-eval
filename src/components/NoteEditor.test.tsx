/* doesnt work */

/*import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { NoteEditor } from './NoteEditor';
import {test, expect, describe, beforeEach} from 'vitest';
import {useStore} from "../../store";


describe('NoteEditor tests', () => {
	beforeEach(() => {
		useStore.getState().reset();
	});

	test('add grade', () => {
		const { addNote } = useStore.getState();
		addNote(
			{title: 'Note 1'},
			{content: ''},
			{comment: 'Comment 1'}
		)
		const { notes } = useStore.getState();
		expect(notes).toMatchSnapshot();

		const updateNote = vi.fn();
		const setShowEditor = vi.fn();
		const index = 0;
		const title = 'Note Title';
		const content = '10';
		const comment = 'Note Comment';

		render(
			<NoteEditor
				index={index}
				title={title}
				content={content}
				comment={comment}
				setShowEditor={setShowEditor}
			/>
		);

		// Empty the title field
		fireEvent.change(screen.getByTestId('title'), { target: { value: '' } });

		// Click the "Sauvegarder" button
		fireEvent.click(screen.getByText('Sauvegarder'));

		// Verify that the updateNote and setShowEditor functions were not called
		expect(updateNote).not.toHaveBeenCalled();
		expect(setShowEditor).not.toHaveBeenCalled();
	});

	test('should update note if all fields are filled', () => {
		useStore.getState().addNote('Note 1', '10', 'Comment 1');
		const updateNote = vi.fn();
		const setShowEditor = vi.fn();
		const index = 0;
		const title = 'Note Title';
		const content = '10';
		const comment = 'Note Comment';

		render(
			<NoteEditor
				index={index}
				title={title}
				content={content}
				comment={comment}
				setShowEditor={setShowEditor}
			/>
		);

		// Click the "Sauvegarder" button
		fireEvent.click(screen.getByText('Sauvegarder'));

		// Verify that the updateNote function was called with the correct arguments
		expect(updateNote).toHaveBeenCalledWith(index, title, content, comment);

		// Verify that the setShowEditor function was called with false
		expect(setShowEditor).toHaveBeenCalledWith(false);
	});
});
*/