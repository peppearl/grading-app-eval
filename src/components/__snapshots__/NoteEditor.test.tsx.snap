// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteEditor tests > should not update note if any field is empty 1`] = `
[
  {
    "comment": {
      "comment": "Comment 1",
    },
    "content": {
      "content": "19",
    },
    "created": "02/06/2023 16:56:49",
    "title": {
      "title": "Note 1",
    },
  },
]
`;
